File: src/Scripting/PHP/patterns/iterator.php
<?php declare(strict_types=1);
$check = static fn(bool $ok) => $ok ?: throw new RuntimeException('iterator');
$collection = new class(['a','b','c']) implements IteratorAggregate {
    public function __construct(private array $items) {}
    public function getIterator(): Traversable { yield from $this->items; }
};
$check(iterator_to_array($collection->getIterator()) === ['a','b','c']);

File: src/Scripting/Lua/patterns/iterator.lua
local collection = { 2, 4, 6 }
local function iterator(items)
  local i = 0
  return function()
    i = i + 1
    return items[i]
  end
end
local next_item = iterator(collection)
assert(next_item() == 2 and next_item() == 4 and next_item() == 6 and next_item() == nil)
return true
